src/functions.js

```diff
@@ -25,144 +41,113 @@ function setFunctionType(block, ...types) {
   block.outputType = outputType;
 }
 
-Blockly.Blocks['even'] = {
-  init: function () {
-    this.appendValueInput("NAME")
-      .appendField("even");
-    this.setInputsInline(true);
-    this.setOutput(true);
-    this.setColour(230);
-    this.setTooltip("");
-    this.setHelpUrl("");
-    this.setOnChange(onChangeFunction.bind(this))
-
-    setFunctionType(this, "Number", "Boolean")
-  }
-}
-
-Blockly.Blocks['not'] = {
-  init: function () {
-    this.appendValueInput("NAME")
-      .appendField("not");
-    this.setInputsInline(true);
-    this.setOutput(true);
-    this.setColour(230);
-    this.setTooltip("");
-    this.setHelpUrl("");
-    this.setOnChange(onChangeFunction.bind(this))
-
-    setFunctionType(this, "Boolean", "Boolean")
-  }
-}
-
-Blockly.Blocks['length'] = {
-  init: function () {
-    this.appendValueInput("NAME")
-      .appendField("length");
-    this.setInputsInline(true);
-    this.setOutput(true);
-    this.setColour(230);
-    this.setTooltip("");
-    this.setHelpUrl("");
-    this.setOnChange(onChangeFunction.bind(this))
-
-    setFunctionType(this, "String", "Number")
+function buildFuctionBlockWith(name, functionType, cb) {
+  Blockly.Blocks[name] = {
+    init: function () {
+      cb(this)
+      this.setInputsInline(true)
+      this.setOutput(true)
+      this.setTooltip("")
+      this.setHelpUrl("")
+      this.setOnChange(onChangeFunction.bind(this))
+      setFunctionType(this, ...functionType)
+    }
   }
-}
-
-Blockly.Blocks['charAt'] = {
-  init: function () {
-    this.appendValueInput("NAME")
-      .appendField("charAt");
-
-    this.appendValueInput("NAME")
 
-    this.setInputsInline(true);
-    this.setOutput(true);
-    this.setColour(230);
-    this.setTooltip("");
-    this.setHelpUrl("");
-    this.setOnChange(onChangeFunction.bind(this))
-
-    setFunctionType(this, "Number", "String", "String")
-  }
 }
 
-Blockly.Blocks['compare'] = {
+const buildFuctionBlock = (name, functionType, fields = []) =>
+  buildFuctionBlockWith(name, functionType, block => {
+    block.appendValueInput(`ARG0`).appendField(fields[0] === undefined ? name : fields[0])
+    for (let index = 1; index < functionType.length - 1; index++) {
+      const inputName = fields[index] || ""
+      block.appendValueInput(`ARG${index}`).appendField(inputName)
+    }
+  })
+
+const buildInfixFuctionBlock = ([name, field], functionType) =>
+  buildFuctionBlockWith(name, functionType, block => {
+    block.appendValueInput("LEFT")
```

Comment:
why are they called LEFT and RIGHT? where do we use that later?

Reply:
Ya estaban así, ahora creo que no se usan pero seguramente lo necesitemos para cuando queramos generar código Haskell. Igual podrían quedar igual que las otras funciones.